### PR TITLE
CDM-425 + CDM-426

### PR DESCRIPTION
**Pull request by costebano**
- fixed state saving;
- added ability to disable target tables that are not connected with current source table

### CODE REVIEW

File: UI/src/app/models/arrow.ts [hunk 1]
```diff
@@ -119,13 +120,15 @@ export class Arrow implements IConnector {
...
 
-  setEndMarkerType(type: string): void {
+  setEndMarkerType(type): void {
     this.refreshPathHtmlElement();
 
     const isActive = this.svgPath.attributes[markerEndAttributeIndex].value.includes('active');
 
     this.renderer.removeAttribute(this.svgPath, 'marker-end');
-    this.renderer.setAttribute(this.svgPath, 'marker-end', `url(#marker-end${isActive ? '-active' : ''}${type !== 'None' ? `-${type}` : ''})`);
+    type = type !== 'None' ? type : '';
```
Comment (etokareva) [inline UI/src/app/models/arrow.ts:129]: I'd preve to check on truth first

File: UI/src/app/components/pages/mapping/mapping.component.ts [hunk 2]
```diff
@@ -270,4 +275,21 @@ export class MappingComponent extends BaseComponent implements OnInit, OnDestroy
       }, 500);
     });
   }
+
+  changeTargetTabIndex() {
+    const tableName = this.source[this.sourceTabIndex].name;
+    const tagretTableNameIndex = 0;
+    const targetTableName = this.mappedTables.find(item => item.includes(tableName))[tagretTableNameIndex];
+    for (let i = 0; i < this.target.length; i++) {
```
Comment (etokareva) [inline UI/src/app/components/pages/mapping/mapping.component.ts:283]: Could we use use `findIndex` method?

File: UI/src/app/models/arrow.ts (hunk 1 above)
Comment (costebano) [reply UI/src/app/models/arrow.ts]: fixed

File: UI/src/app/components/pages/mapping/mapping.component.ts (hunk 2 above)
Comment (costebano) [reply UI/src/app/components/pages/mapping/mapping.component.ts]: fixed

File: UI/src/app/models/arrow.ts [hunk 3]
```diff
@@ -119,13 +120,15 @@ export class Arrow implements IConnector {
...
-  setEndMarkerType(type: string): void {
+  setEndMarkerType(type): void {
     this.refreshPathHtmlElement();
 
     const isActive = this.svgPath.attributes[markerEndAttributeIndex].value.includes('active');
 
     this.renderer.removeAttribute(this.svgPath, 'marker-end');
-    this.renderer.setAttribute(this.svgPath, 'marker-end', `url(#marker-end${isActive ? '-active' : ''}${type !== 'None' ? `-${type}` : ''})`);
+    type = type === 'None' ? '' : type;
+    this.renderer.setAttribute(this.svgPath, 'marker-end', `url(#marker-end${isActive ? '-active' : ''}${type ? `-${type}` : ''})`);
```
Comment (etokareva) [inline UI/src/app/models/arrow.ts:130]: Is this `url(#marker-end${isActive ? '-active' : ''}${type ? ` unclosed type variable?

File: UI/src/app/services/bridge.service.ts [hunk 4]
```diff
@@ -250,6 +253,7 @@ export class BridgeService {
   setArrowType(id: string, type: string) {
     const arrow = this.arrowsCache[id];
     arrow.connector.setEndMarkerType(type);
+    arrow.type = type !== 'None' ? type : '';
```
[the remark this answers is not in the document]
Comment (costebano) [reply UI/src/app/services/bridge.service.ts]: fixed

File: UI/src/app/models/arrow.ts (hunk 3 above)
Comment (etokareva) [reply UI/src/app/models/arrow.ts:130]: Shouldn't it be 
```ts
this.renderer.setAttribute(this.svgPath, 'marker-end', `url(#marker-end${isActive ? '-active' : ''}${type} ? `-${type}` : ''})`);
```
?

Probably it's better to split the logic of checking these properties.
```ts
const markerActive= isActive ? '-active' : '';
const markerType= type ? `-${type}` : '';
this.renderer.setAttribute(this.svgPath, 'marker-end', `url(#marker-end${markerActive}${markerType})`);
```

Comment (costebano) [reply UI/src/app/models/arrow.ts]: refactored to make logic clear

**Pull request merged**